### PR TITLE
chore: adding tests for composed disposable token scopes

### PR DESCRIPTION
**Pull request by pgautier404**
This commit adds an assortment of tests against `DisposableTokenScope` objects that are composed of multiple permissions to supplement the existing tests which rely on predefined single-permission `DisposableTokenScopes` objects. Similar tests covering topics are being handled in a separate PR that adds token scoping by topic name prefix.

### CODE REVIEW

File: tests/Integration/Momento.Sdk.Tests/AuthClientCacheTest.cs [hunk 1]
```diff
@@ -553,4 +550,173 @@ public async Task GenerateDisposableCacheKeyPrefixAuthToken_WriteOnly_HappyPath(
...
+        Assert.True(getResponse is CacheGetResponse.Hit);
+        getResponse = await client.GetAsync(cache2Name, "pet-koala");
+        Assert.True(getResponse is CacheGetResponse.Hit);
+        getResponse = await client.GetAsync(cacheName, "dog");
+        AssertPermissionError<CacheGetResponse, CacheGetResponse.Error>(getResponse);
+        getResponse = await client.GetAsync(cache2Name, "dog");
+        AssertPermissionError<CacheGetResponse, CacheGetResponse.Error>(getResponse);
+
+        // delete the second cache
+        Assert.True(await cacheClient.DeleteCacheAsync(cache2Name) is DeleteCacheResponse.Success);
```
Comment (nand4011) [inline tests/Integration/Momento.Sdk.Tests/AuthClientCacheTest.cs:672]: LGTM, but these should go in a finally so we don't leak caches if the test fails.

Comment (pgautier404) [reply tests/Integration/Momento.Sdk.Tests/AuthClientCacheTest.cs:672]: Totally, great point, thanks.

Comment (pgautier404) [reply tests/Integration/Momento.Sdk.Tests/AuthClientCacheTest.cs]: Fixed in 3a6928b.

**Review by anitarua — APPROVED** [review]
lgtm!

**Pull request merged**